src/Tools/Performance Toolkit/App/src/BCPTTestSuite.Codeunit.al

```diff
@@ -411,4 +411,20 @@ codeunit 149006 "BCPT Test Suite"
         BCPTLine.SetRange("BCPT Code", SuiteCode);
         BCPTLine.SetRange("Codeunit ID", CodeunitID);
     end;
+
+    /// <summary>
+    /// This event is raised before a log entry is added to the BCPT Line table.
+    /// It can be used to skip errors which are not relevant for the test suite. Like unused handler functions.
+    /// </summary>
+    /// <param name="SuiteCode">The test suite code</param>
+    /// <param name="CodeunitId">The id of the test codunit that is being run</param>
+    /// <param name="Description">Description of the test on the "BCPT Line"</param>
+    /// <param name="Operation">Which operation that is currently executed</param>
+    /// <param name="ExecutionSuccess">If the test succedded or not</param>
+    /// <param name="Message"></param>
+    /// <param name="IsHandled">Handled pattern parameter</param>
+    [IntegrationEvent(false, false)]
+    procedure OnBeforeBCPTLineAddLogEntry(SuiteCode: Code[10]; CodeunitId: Integer; Description: Text; var Operation: Text; var ExecutionSuccess: Boolean; var Message: Text; var IsHandled: Boolean)
```

Comment:
why are any of these parameter var? As far as i can tell, the only way it could have an effect is if you changed the values but did not set IsHandled. If you did that you would change what is logged. If you set IsHandled you would simply not log anything and changing the values would not have any effect. 

If what you want is to change the values that are written to the log then I think there should be an indication that the log was altered by a subscriber.

Reply:
I have been running this code local for while but it was not until now I realised that that skipping this function is not a good scenario because what we want to achieve is to change from a "faulty log" to a correct log. Which means changing the values without setting IsHandled. Otherwise we won't receive a complete log for that scenario 😏

Example: 
Three scenarios:
1. Is having a unused handled event which gives error at the end of execution
2. Is having a handled unused handled event which wont't give an error at the end of execution
3. A normal test that completes successfully for comparison

The event code:
![image](https://github.com/microsoft/BCApps/assets/34970096/8552358c-a7b7-4672-a2e8-00ed4d8d6ad1)

Result:
![image](https://github.com/microsoft/BCApps/assets/34970096/b3dc12bb-9041-4cc0-9631-ccbd9ea3a460)

What do you recommend as notifier for changed values? A boolean?

Reply:
Ok. Do you still need the IsHandled to skip writing the log completely? Wouldn't you always want a log entry? 

Regarding the ValuesAreChanged Boolean, I think that should be automatic. It should not require anything from the subscriber. So, you would need to save the values before the event, compare after the event, and if they are changed then write that to the log. You might consider passing the original values to the event as non-var and the new ones as var. That way if there are multiple subscribers they can check whether the log was already modified when they were called.

You could use a boolean, but given that you need to old and new values anyway to check, you could just add fields to the log entry like 
"Orig. Operation"
"Orig. Message"
"Orig. Exec. Status"

now the subscriber can edit the values and all will be recorded so you'll be able to track the changes in the log.

Reply:
Have added these parameters, plus initialization and handling of them.